core/src/main/java/com/linecorp/armeria/client/RefreshingAddressResolverGroup.java

```diff
@@ -117,9 +132,15 @@
         }
     }
 
+    @Nullable
     @VisibleForTesting
-    ConcurrentMap<String, CompletableFuture<CacheEntry>> cache() {
-        return cache;
+    Cache<String, CompletableFuture<CacheEntry>> cache() {
+        return dnsCache;
+    }
+
+    @VisibleForTesting
+    void cache(@Nullable Cache<String, CompletableFuture<CacheEntry>> dnsCache) {
```

Comment:
If we add the setter to `DnsResolverGroupBuilder`, we don't need this method.
We can accept the cache spec in the constructor so that we can make `dnsCache` as a final field.